test/wasm2c/minimal.txt

```diff
@@ -532,14 +536,15 @@ static inline void memory_copy(wasm_rt_memory_t* dest,
 }
 
 static inline void memory_init(wasm_rt_memory_t* dest,
+                               bool check_only,
                                const u8* src,
                                u32 src_size,
                                u32 dest_addr,
                                u32 src_addr,
                                u32 n) {
   if (UNLIKELY(src_addr + (uint64_t)n > src_size))
     TRAP(OOB);
-  LOAD_DATA((*dest), dest_addr, src + src_addr, n);
+  LOAD_DATA((*dest), dest_addr, check_only, src + src_addr, n);
```

Comment:
Rather than add an extra argument to the LOAD_DATA macro could we define a separate `memory_check` or do:

```
```
if (check_only)
  RANGE_CHECK
else
  LOAD_DATA
```

Better still, should we just use `RANGE_CHECK` here since IIRC we are not actually trying to load anything.

Actually, maybe that doesn't work since RANGE_CHECK doesn't always do anything... maybe we need something like `FORCE_RANGE_CHECK` here?

Reply:
What makes this annoying is that the range that LOAD_DATA needs to check depends (I think) on the endianness -- it's written differently in the different cases. (I'm not 100% confident wasm2c currently works on little-endian platforms, but I think we still nominally care about doing the right thing there.)

How do you feel about the version I just pushed, where we just completely eliminate the check_only parameter from everywhere, and instead we just test "n" in load_data()? This has a tiny runtime cost but I'm guessing it's in the noise in any realistic setting, and it's a lot cleaner/simpler.

(BTW RANGE_CHECK always checks -- it's MEMCHECK that turns into a no-op when using the signal handler.)